sqlitecluster/SQLiteNode.cpp

```diff
@@ -2241,30 +2384,24 @@ void SQLiteNode::handleBeginTransaction(Peer* peer, const SData& message) {
         string verb = success ? "APPROVE_TRANSACTION" : "DENY_TRANSACTION";
         if (!SStartsWith(message["ID"], "ASYNC_")) {
             // Not a permafollower, approve the transaction
-            PINFO(verb << " #" << _db.getCommitCount() + 1 << " (" << message["NewHash"] << ").");
+            PINFO(verb << " #" << db.getCommitCount() + 1 << " (" << message["NewHash"] << ").");
             SData response(verb);
-            response["NewCount"] = SToStr(_db.getCommitCount() + 1);
-            response["NewHash"] = success ? _db.getUncommittedHash() : message["NewHash"];
+            response["NewCount"] = SToStr(db.getCommitCount() + 1);
+            response["NewHash"] = success ? db.getUncommittedHash() : message["NewHash"];
             response["ID"] = message["ID"];
+
+            lock_guard<mutex> leadPeerLock(_leadPeerMutex);
+            if (!_leadPeer) {
```

Comment:
No, if we've lost leader during `beginTransaction` we shouldn't die. This didn't used to be possible, but now that leader changes state in a different thread, it could happen.